src/main/java/com/google/devtools/build/lib/syntax/SkylarkDict.java

```diff
@@ -355,7 +355,7 @@ public void unsafeShallowFreeze() {
    */
   public void put(K key, V value, Location loc, Mutability mutability) throws EvalException {
     checkMutable(loc, mutability);
-    EvalUtils.checkValidDictKey(key);
+    EvalUtils.checkValidDictKey(key, null);
```

Comment:
I think that this function (put) is only used here: https://github.com/bazelbuild/bazel/blob/master/src/main/java/com/google/devtools/build/lib/analysis/skylark/FunctionTransitionUtil.java#L178
However, I couldn't find a way to get environment here and passing null will just assume the flag is false.